### ISSUE TITLE
screengrab Unable to get a screenshot storage directory

### ISSUE THREAD

**Issue by NicholasIoanJones** [fastlane/fastlane#17164]
- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Running screengrab through fastlane but getting "Unable to capture screenshot" and "Unable to get a screenshot storage directory"

##### Command executed
bundle exec fastlane screenshots

### Captured Output

Command Used: `screenshots --capture_output`
<details><summary>Output/Log</summary>

```

DEBUG [2020-09-04 13:24:43.07]: Checking if there are any plugins that should be loaded...
DEBUG [2020-09-04 13:24:43.07]: Loading 'fastlane-plugin-appicon' plugin
INFO [2020-09-04 13:24:43.07]: gem 'fastlane-plugin-appicon' is already installed


```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.6                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | ~/.gems/bin/fastlane                                           |
| Host                        | Mac OS X 10.15.6 (19G2021)                                     |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.7                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:android)

platform :android do

  desc "Generate all correct icon sizes from source"
  lane :icons do
    android_appicon(
      appicon_image_file: 'AppIcon/IconAndroid.png',
      appicon_path: './app/src/main/res/mipmap'
    )
  end

  desc "Runs all the tests"
  lane :test do
    gradle(task: "clearData")
    gradle(task: "test")
  end

  desc "Capture a new set of screenshots"
  lane :screenshots do
    # gradle(task: "assembleDebug assembleAndroidTest")
    # gradle(task: "clearData")
    screengrab(
        clear_previous_screenshots: true,
        test_instrumentation_runner: 'androidx.test.runner.AndroidJUnitRunner',
        app_apk_path: "./app/build/outputs/apk/debug/app-debug.apk",
        tests_apk_path: "./app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk",
        # Note: Since we need to reset the app data between the tests we cannot supply multiple locales
        #        here. As a workaround just call screengrab multiple times.
        locales: ["en-US"]
    )
  end

  desc "Submit a new build to the Google Play store on alpha track and then promote to beta track (closed testing)"
  lane :beta do
    gradle(task: "clean bundleRelease")
    supply(
        aab: "./app/build/outputs/bundle/release/app-release.aab",
        track: "alpha",
    )
    supply(
        track: "alpha",
        track_promote_to: "beta",
        skip_upload_apk: true,
        skip_upload_aab: true,
        skip_upload_images: true,
        skip_upload_metadata: true,
        skip_upload_changelogs: true,
        skip_upload_screenshots: true,
    )
  end

  desc "Deploy a new version to Google Play (by promoting the build currently in the beta track)"
  lane :deploy do
    supply(
        track: "beta",
        track_promote_to: "production",
        skip_upload_apk: true,
        skip_upload_aab: true,
        skip_upload_images: true,
        skip_upload_metadata: true,
        skip_upload_changelogs: true,
        skip_upload_screenshots: true,
    )
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file "./secrets.json" # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
package_name "com.txpchat.txpchatv2"

```
</details>

### fastlane gems

| Gem        | Version | Update-Status |
| ---------- | ------- | ------------- |
| fastlane   | 2.157.4 | ✅ Up-To-Date  |
| screengrab | 1.0.0   | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status |
| ----------------------- | ------- | ------------- |
| fastlane-plugin-appicon | 0.15.0  | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| bundler                   | 2.0.2        |
| rake                      | 13.0.1       |
| CFPropertyList            | 3.0.2        |
| public_suffix             | 4.0.6        |
| addressable               | 2.7.0        |
| atomos                    | 0.1.3        |
| aws-eventstream           | 1.1.0        |
| aws-partitions            | 1.364.0      |
| aws-sigv4                 | 1.2.2        |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.105.0      |
| aws-sdk-kms               | 1.37.0       |
| aws-sdk-s3                | 1.79.1       |
| babosa                    | 1.0.3        |
| claide                    | 1.0.3        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.6.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.6        |
| emoji_regex               | 3.0.0        |
| excon                     | 0.76.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 1.0.1        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.7        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.2.2        |
| memoist                   | 0.16.2       |
| multi_json                | 1.15.0       |
| os                        | 1.1.1        |
| signet                    | 0.14.0       |
| googleauth                | 0.13.1       |
| httpclient                | 2.8.3        |
| mini_mime                 | 1.0.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.3        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.28.0       |
| json                      | 2.3.1        |
| mini_magick               | 4.10.1       |
| plist                     | 3.5.0        |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.3.0        |
| xcodeproj                 | 1.18.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-appicon   | 0.15.0       |
</details>

*generated on:* **2020-09-04**
</details>

[✔] 🚀 
+-------------------------+---------+-------------------------+
|                        Used plugins                         |
+-------------------------+---------+-------------------------+
| Plugin                  | Version | Action                  |
+-------------------------+---------+-------------------------+
| fastlane-plugin-appicon | 0.15.0  | android_appicon appicon |
+-------------------------+---------+-------------------------+

[13:25:20]: ------------------------------
[13:25:20]: --- Step: default_platform ---
[13:25:20]: ------------------------------
[13:25:20]: Driving the lane 'android screenshots' 🚀
[13:25:20]: ------------------------
[13:25:20]: --- Step: screengrab ---
[13:25:20]: ------------------------

+-----------------------------+---------------------------------------------------------------------+
|                                   Summary for screengrab 1.0.0                                    |
+-----------------------------+---------------------------------------------------------------------+
| clear_previous_screenshots  | true                                                                |
| test_instrumentation_runner | androidx.test.runner.AndroidJUnitRunner                             |
| app_apk_path                | ./app/build/outputs/apk/debug/app-debug.apk                         |
| tests_apk_path              | ./app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk |
| locales                     | ["en-US"]                                                           |
| tests_package_name          | com.txpchat.txpchatv2.test                                          |
| android_home                | /Users/nickjones/Library/Android/sdk                                |
| output_directory            | fastlane/metadata/android                                           |
| skip_open_summary           | false                                                               |
| app_package_name            | com.txpchat.txpchatv2                                               |
| ending_locale               | en-US                                                               |
| device_type                 | phone                                                               |
+-----------------------------+---------------------------------------------------------------------+

[13:25:20]: Limiting the test classes run by `screengrab` to just those that generate screenshots can make runs faster.
[13:25:20]: Consider using the :use_tests_in_classes or :use_tests_in_packages option, and organize your tests accordingly.
[13:25:20]: Clearing phoneScreenshots within fastlane/metadata/android
[13:25:20]: $ adb devices -l
[13:25:20]: ▸ List of devices attached
[13:25:20]: ▸ emulator-5554          device product:sdk_gphone_x86 model:Android_SDK_built_for_x86 device:generic_x86 transport_id:1
[13:25:20]: $ adb -s emulator-5554 shell echo \$EXTERNAL_STORAGE
[13:25:20]: ▸ /sdcard
[13:25:20]: Cleaning screenshots on device
ls: /sdcard/com.txpchat.txpchatv2/screengrab: No such file or directory
[13:25:21]: Exit status: 1
ls: /data/data/com.txpchat.txpchatv2/app_screengrab: Permission denied
[13:25:21]: Exit status: 1
[13:25:21]: Validating app APK
[13:25:21]: $ /Users/nickjones/Library/Android/sdk/build-tools/30.0.2/aapt dump permissions ./app/build/outputs/apk/debug/app-debug.apk
[13:25:21]: ▸ package: com.txpchat.txpchatv2
[13:25:21]: ▸ uses-permission: name='android.permission.DISABLE_KEYGUARD'
[13:25:21]: ▸ uses-permission: name='android.permission.WAKE_LOCK'
[13:25:21]: ▸ uses-permission: name='android.permission.WRITE_EXTERNAL_STORAGE'
[13:25:21]: ▸ uses-permission: name='android.permission.READ_EXTERNAL_STORAGE'
[13:25:21]: ▸ uses-permission: name='android.permission.CHANGE_CONFIGURATION'
[13:25:21]: ▸ uses-permission: name='android.permission.INTERNET'
[13:25:21]: ▸ uses-permission: name='android.permission.SYSTEM_ALERT_WINDOW'
[13:25:21]: ▸ uses-permission: name='android.permission.FOREGROUND_SERVICE'
[13:25:21]: ▸ uses-permission: name='android.permission.READ_CONTACTS'
[13:25:21]: ▸ uses-permission: name='android.permission.READ_PROFILE'
[13:25:21]: ▸ uses-permission: name='android.permission.CAMERA'
[13:25:21]: ▸ uses-permission: name='android.permission.RECEIVE_BOOT_COMPLETED'
[13:25:21]: ▸ uses-permission: name='android.permission.VIBRATE'
[13:25:21]: ▸ uses-permission: name='android.permission.USE_BIOMETRIC'
[13:25:21]: ▸ uses-permission: name='android.permission.USE_FINGERPRINT'
[13:25:21]: ▸ uses-permission: name='android.permission.ACCESS_WIFI_STATE'
[13:25:21]: ▸ uses-permission: name='android.permission.ACCESS_NETWORK_STATE'
[13:25:21]: ▸ uses-permission: name='com.android.vending.CHECK_LICENSE'
[13:25:21]: ▸ uses-permission: name='com.google.android.c2dm.permission.RECEIVE'
[13:25:21]: ▸ uses-permission: name='com.sec.android.provider.badge.permission.READ'
[13:25:21]: ▸ uses-permission: name='com.sec.android.provider.badge.permission.WRITE'
[13:25:21]: ▸ uses-permission: name='com.htc.launcher.permission.READ_SETTINGS'
[13:25:21]: ▸ uses-permission: name='com.htc.launcher.permission.UPDATE_SHORTCUT'
[13:25:21]: ▸ uses-permission: name='com.sonyericsson.home.permission.BROADCAST_BADGE'
[13:25:21]: ▸ uses-permission: name='com.sonymobile.home.permission.PROVIDER_INSERT_BADGE'
[13:25:21]: ▸ uses-permission: name='com.anddoes.launcher.permission.UPDATE_COUNT'
[13:25:21]: ▸ uses-permission: name='com.majeur.launcher.permission.UPDATE_BADGE'
[13:25:21]: ▸ uses-permission: name='com.huawei.android.launcher.permission.CHANGE_BADGE'
[13:25:21]: ▸ uses-permission: name='com.huawei.android.launcher.permission.READ_SETTINGS'
[13:25:21]: ▸ uses-permission: name='com.huawei.android.launcher.permission.WRITE_SETTINGS'
[13:25:21]: ▸ uses-permission: name='android.permission.READ_APP_BADGE'
[13:25:21]: ▸ uses-permission: name='com.oppo.launcher.permission.READ_SETTINGS'
[13:25:21]: ▸ uses-permission: name='com.oppo.launcher.permission.WRITE_SETTINGS'
[13:25:21]: ▸ uses-permission: name='me.everything.badger.permission.BADGE_COUNT_READ'
[13:25:21]: ▸ uses-permission: name='me.everything.badger.permission.BADGE_COUNT_WRITE'
[13:25:21]: ▸ uses-permission: name='com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE'
[13:25:21]: Installing app APK
[13:25:21]: $ adb -s emulator-5554 install -r ./app/build/outputs/apk/debug/app-debug.apk
[13:25:21]: ▸ Performing Streamed Install
[13:25:25]: ▸ Success
[13:25:25]: Installing tests APK
[13:25:25]: $ adb -s emulator-5554 install -r ./app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk
[13:25:26]: ▸ Performing Streamed Install
[13:25:26]: ▸ Success
[13:25:26]: Granting the permission necessary to change locales on the device
[13:25:26]: $ adb -s emulator-5554 shell pm grant com.txpchat.txpchatv2 android.permission.CHANGE_CONFIGURATION
[13:25:26]: $ adb -s emulator-5554 shell getprop ro.build.version.sdk
[13:25:26]: ▸ 29
[13:25:26]: Granting the permissions necessary to access device external storage
[13:25:26]: $ adb -s emulator-5554 shell pm grant com.txpchat.txpchatv2 android.permission.WRITE_EXTERNAL_STORAGE
[13:25:26]: $ adb -s emulator-5554 shell pm grant com.txpchat.txpchatv2 android.permission.READ_EXTERNAL_STORAGE
[13:25:26]: Running tests for locale: en-US
[13:25:26]: $ adb -s emulator-5554 shell am instrument --no-window-animation -w \
-e testLocale en_US \
-e endingLocale en_US \
com.txpchat.txpchatv2.test/androidx.test.runner.AndroidJUnitRunner
[13:25:39]: ▸ com.txpchat.txpchatv2.MainActivityTest:
[13:25:39]: ▸ Error in MainActivityTest(com.txpchat.txpchatv2.MainActivityTest):
[13:25:39]: ▸ java.lang.RuntimeException: Unable to capture screenshot.
[13:25:39]: ▸   at tools.fastlane.screengrab.FileWritingScreenshotCallback.screenshotCaptured(FileWritingScreenshotCallback.java:56)
[13:25:39]: ▸   at tools.fastlane.screengrab.UiAutomatorScreenshotStrategy.takeScreenshot(UiAutomatorScreenshotStrategy.java:39)
[13:25:39]: ▸   at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:94)
[13:25:39]: ▸   at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:76)
[13:25:39]: ▸   at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:60)
[13:25:39]: ▸   at com.txpchat.txpchatv2.MainActivityTest.MainActivityTest(MainActivityTest.java:79)
[13:25:39]: ▸   at java.lang.reflect.Method.invoke(Native Method)
[13:25:39]: ▸   at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
[13:25:39]: ▸   at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[13:25:39]: ▸   at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
[13:25:39]: ▸   at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[13:25:39]: ▸   at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
[13:25:39]: ▸   at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
[13:25:39]: ▸   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
[13:25:39]: ▸   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[13:25:39]: ▸   at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[13:25:39]: ▸   at tools.fastlane.screengrab.locale.LocaleTestRule$1.evaluate(LocaleTestRule.java:32)
[13:25:39]: ▸   at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[13:25:39]: ▸   at androidx.test.runner.AndroidJUnit4.run(AndroidJUnit4.java:104)
[13:25:39]: ▸   at org.junit.runners.Suite.runChild(Suite.java:128)
[13:25:39]: ▸   at org.junit.runners.Suite.runChild(Suite.java:27)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[13:25:39]: ▸   at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
[13:25:39]: ▸   at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
[13:25:39]: ▸   at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
[13:25:39]: ▸   at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:388)
[13:25:39]: ▸   at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2189)
[13:25:39]: ▸ Caused by: java.io.IOException: Unable to get a screenshot storage directory
[13:25:39]: ▸   at tools.fastlane.screengrab.FileWritingScreenshotCallback.getFilesDirectory(FileWritingScreenshotCallback.java:81)
[13:25:39]: ▸   at tools.fastlane.screengrab.FileWritingScreenshotCallback.screenshotCaptured(FileWritingScreenshotCallback.java:39)
[13:25:39]: ▸   ... 38 more
[13:25:39]: ▸ Time: 11.367
[13:25:39]: ▸ There was 1 failure:
[13:25:39]: ▸ 1) MainActivityTest(com.txpchat.txpchatv2.MainActivityTest)
[13:25:39]: ▸ java.lang.RuntimeException: Unable to capture screenshot.
[13:25:39]: ▸   at tools.fastlane.screengrab.FileWritingScreenshotCallback.screenshotCaptured(FileWritingScreenshotCallback.java:56)
[13:25:39]: ▸   at tools.fastlane.screengrab.UiAutomatorScreenshotStrategy.takeScreenshot(UiAutomatorScreenshotStrategy.java:39)
[13:25:39]: ▸   at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:94)
[13:25:39]: ▸   at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:76)
[13:25:39]: ▸   at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:60)
[13:25:39]: ▸   at com.txpchat.txpchatv2.MainActivityTest.MainActivityTest(MainActivityTest.java:79)
[13:25:39]: ▸   at java.lang.reflect.Method.invoke(Native Method)
[13:25:39]: ▸   at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
[13:25:39]: ▸   at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[13:25:39]: ▸   at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
[13:25:39]: ▸   at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[13:25:39]: ▸   at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
[13:25:39]: ▸   at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
[13:25:39]: ▸   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
[13:25:39]: ▸   at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[13:25:39]: ▸   at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[13:25:39]: ▸   at tools.fastlane.screengrab.locale.LocaleTestRule$1.evaluate(LocaleTestRule.java:32)
[13:25:39]: ▸   at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[13:25:39]: ▸   at androidx.test.runner.AndroidJUnit4.run(AndroidJUnit4.java:104)
[13:25:39]: ▸   at org.junit.runners.Suite.runChild(Suite.java:128)
[13:25:39]: ▸   at org.junit.runners.Suite.runChild(Suite.java:27)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[13:25:39]: ▸   at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[13:25:39]: ▸   at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[13:25:39]: ▸   at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
[13:25:39]: ▸   at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
[13:25:39]: ▸   at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
[13:25:39]: ▸   at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:388)
[13:25:39]: ▸   at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2189)
[13:25:39]: ▸ Caused by: java.io.IOException: Unable to get a screenshot storage directory
[13:25:39]: ▸   at tools.fastlane.screengrab.FileWritingScreenshotCallback.getFilesDirectory(FileWritingScreenshotCallback.java:81)
[13:25:39]: ▸   at tools.fastlane.screengrab.FileWritingScreenshotCallback.screenshotCaptured(FileWritingScreenshotCallback.java:39)
[13:25:39]: ▸   ... 38 more
[13:25:39]: ▸ FAILURES!!!
[13:25:39]: ▸ Tests run: 1,  Failures: 1
+------------------+---------------------+
|              Lane Context              |
+------------------+---------------------+
| DEFAULT_PLATFORM | android             |
| PLATFORM_NAME    | android             |
| LANE_NAME        | android screenshots |
+------------------+---------------------+
[13:25:40]: Tests failed

+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 💥   | screengrab       | 19          |
+------+------------------+-------------+

[13:25:40]: fastlane finished with errors

[!] Tests failed

**Comment by fatihyildizhan** [comment]
There is a similar problem with iOS Snapshot. https://github.com/fastlane/fastlane/issues/17039

**Comment by lolucosmin** [comment]
Similar, contains also the reproduced steps(config): #17084

**Comment by tmaly1980** [comment]
I'm experiencing this problem at least on CircleCI, with zero explanation. Any ideas? Is this due to changes in fastlane?

**Comment by blastrock** [comment]
Same problem here, found a workaround: https://stackoverflow.com/a/56809469/1810193

tl;dr: Set a target API level < 29

**Comment by lolucosmin** [comment]
The problem is still persisting.

**Comment by rogerluan** [comment]
> Same problem here, found a workaround: https://stackoverflow.com/a/56809469/1810193
> 
> tl;dr: Set a target API level < 29

Did this work for you @lolucosmin ?

**Comment by 1951FDG** [comment]
A better workaround, until Fastlane fixes this issue:

If you target SDK 29 or higher you should add `android:requestLegacyExternalStorage="true"` to the application definition in the debug AndroidManifest.xml only! (e.g., app/src/debug/AndroidManifest.xml)

```xml
<manifest ... >
  <!-- This attribute is "false" by default on apps targeting
       Android 10 or higher. -->
    <application android:requestLegacyExternalStorage="true" ... >
      ...
    </application>
</manifest>
```

The fastlane-bot is aggressive! Issues like these are getting closed too soon!
